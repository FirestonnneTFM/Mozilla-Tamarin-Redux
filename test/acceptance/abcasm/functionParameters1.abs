.script_info { 
	init: .function_id(main)
	traits: {
		.trait { kind: Trait_Slot name:oneParam }
		.trait { kind: Trait_Slot name:intFunc }
	}
}

function main()
{
	getlocal0
	pushscope

	findproperty oneParam
	newfunction .function_id(oneParam)
	setslot .slot_id(oneParam)

	findproperty intFunc
	newfunction .function_id(intFunc)
	setslot .slot_id(intFunc)

	getlocal0
	pushstring "Param one"
	callproperty oneParam 1


	getlocal0
	pushbyte 4
	callproperty intFunc 1
	.print

	returnvoid
}


//function oneParam(String)
.method_info { name:oneParam return_type:void param_type: {String} }

.method_body_info {
	method: .function_id(oneParam)
	code:
	{
		getlocal1
		.print
		returnvoid
	}
}

function intFunc(int):int
{
	pushdouble 3.14157
	getlocal1
	add
	dup
	.print
	returnvalue
}
