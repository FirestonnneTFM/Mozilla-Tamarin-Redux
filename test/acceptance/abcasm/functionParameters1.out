Param one
7.14157
7
